ompi/mca/coll/han/coll_han_gatherv.c

```diff
@@ -332,7 +332,7 @@ int mca_coll_han_gatherv_intra(const void *sbuf, int scount, struct ompi_datatyp
     }
 
     if (max_data_size > (size_t) INT_MAX) {
-        datatype_size = coll_han_utils_gcd(low_data_size, low_size);
+        datatype_size = coll_han_utils_gcd((const size_t *) low_data_size, low_size);
```

Comment:
This will silence the warning but is not necessarily a fix. I think the safer way to go is to have a temporary `size_t` that is assigned to `low_data_size` after the call.

Reply:
I was being lazy since this code will go away after we embiggen the collectives. I will put in the fix as suggested.